### PR TITLE
Add support for template to nav-item

### PR DESCRIPTION
**Pull request by awinny**
- Added support for providing a html template to be added dynamically when using data to generate nav-items
- Added fullscreen component as a demonstration of template support in nav-item
- Added support for setting classes to nav-item by providing classes prop
- Refactored nav-item and demo page

### CODE REVIEW

File: demo/examples/more-dynamic2.html [hunk 1]
```diff
@@ -61,248 +63,250 @@ <h1>headline</h1>
...
         {
-          text:'first', location:'#/first'
+          text:'First', location:'#/First'
         }
       ];
 
       this.primaryItems = [
         {
-          text:'first', location:'#/first', caption: 'first', active: true,
+          text:'First', location:'#/First', caption: 'First', active: true, classes: 'first-item',
```
Comment (awinny) [inline demo/examples/more-dynamic2.html]: Here we use new classes prop

File: demo/examples/more-dynamic2.html [hunk 2]
```diff
@@ -61,248 +63,250 @@ <h1>headline</h1>
...
-            { text:'fourth', location:'#/first/fourth' },
-            { text:'fifth', location:'#/first/fifth' },
-            { text:'sixth', location:'#/first/sixth' }
+            { text:'First', location:'#/First/First', active: true },
+            { text:'Second', location:'#/First/Second' },
+            { text:'Third', location:'#/First/Third' },
+            { text:'Fourth', location:'#/First/Fourth' },
+            { text:'Fifth', location:'#/First/Fifth' },
+            { text:'Sixth', location:'#/First/Sixth' },
+            { template:'<c-fullscreen></c-fullscreen>' }
```
Comment (awinny) [inline demo/examples/more-dynamic2.html]: Here we use the new template prop

File: demo/examples/more-dynamic2.html [hunk 3]
```diff
@@ -61,248 +63,250 @@ <h1>headline</h1>
...
+            { text:'First', location:'#/end/First' },
+            { text:'Second', location:'#/end/Second' },
+            { text:'Third', location:'#/end/Third' },
+            { text:'Fourth', location:'#/end/Fourth' },
+            { text:'Fifth', location:'#/end/Fifth' },
+            { text:'Sixth', location:'#/end/Sixth' }
           ]
-        }
+        },
+        { template:'<c-fullscreen></c-fullscreen>', classes: 'pull-right' }
```
Comment (awinny) [inline demo/examples/more-dynamic2.html]: Here we use both the new template and classes prop

File: src/components/main-navigation/template.html [hunk 4]
```diff
@@ -3,45 +3,18 @@
...
-          <template is="dom-if" if="{{item.children}}">
-            <sub-navigation caption$="{{item.caption}}">
-              <template is="dom-repeat" items="{{item.children}}" as="subItem">
-                <nav-item location$="{{subItem.location}}" active$="{{subItem.active}}" text$="{{subItem.text}}" class$="{{dashed(subItem.text)}}"></nav-item>
-              </template>
-            </sub-navigation>
-          </template>
-        </nav-item> -->
+      <template is="dom-repeat" items="{{primaryItems}}">
+        <nav-item text="{{item.text}}" template="{{item.template}}" location="{{item.location}}" active="{{item.active}}" children="{{item.children}}" classes="{{item.classes}}" class$="{{dashed(item.text)}}" caption="{{item.text}}"></nav-item>
```
Comment (awinny) [inline src/components/main-navigation/template.html]: Here we set template and classes prop if they exist

File: src/components/main-navigation/template.html [hunk 5]
```diff
@@ -3,45 +3,18 @@
...
-      </div> -->
     </primary-items>
 
     <secondary-items>
       <content select="secondary-items" id="secondary-items"></content>
 
-      <template is="dom-repeat" items="{{secondaryItems}}" sort="sort">
-        <nav-item location="{{item.location}}" text="{{item.text}}" class$="dropdown {{dashed(item.text)}}" active="{{item.active}}" children="{{item.children}}" dropdown="{{item.children.length}}"></nav-item>
+      <template is="dom-repeat" items="{{secondaryItems}}">
+        <nav-item text="{{item.text}}" template="{{item.template}}" location="{{item.location}}" active="{{item.active}}" children="{{item.children}}" classes="{{item.classes}}" class$="dropdown {{dashed(item.text)}}" dropdown="{{item.children.length}}"></nav-item>
```
Comment (awinny) [inline src/components/main-navigation/template.html]: Here we set template and classes prop if they exist

Comment (fomekh) [reply src/components/main-navigation/template.html]: Good now

Comment (scania-bootstrap) [reply src/components/main-navigation/template.html]: Wow. Documentation of how to use this one?

**Pull request merged**